pandas/core/ops.py

```diff
@@ -1035,14 +1059,21 @@ def safe_na_op(lvalues, rvalues):
             raise
 
     def wrapper(left, right):
-
         if isinstance(right, ABCDataFrame):
             return NotImplemented
 
         left, right = _align_method_SERIES(left, right)
         res_name = get_op_result_name(left, right)
 
-        if is_datetime64_dtype(left) or is_datetime64tz_dtype(left):
+        if is_categorical_dtype(left):
```

Comment:
This will be made cleaner if/when we implement these ops in Categorical itself (they'll just raise the TypeError below)